src/main/java/picard/sam/AbstractAlignmentMerger.java

```diff
@@ -742,28 +756,55 @@ private void transferAlignmentInfoToPairedRead(final SAMRecord firstUnaligned, f
             SamPairUtil.setProperPairFlags(secondUnaligned, firstUnaligned, expectedOrientations);
         }
     }
 
+
+
     /**
-     * Checks to see whether the ends of the reads overlap and soft clips reads
-     * them if necessary.
+     * Checks to see whether the ends of the reads overlap and clips reads
+     * if necessary.
      */
-    protected static void clipForOverlappingReads(final SAMRecord read1, final SAMRecord read2) {
+    protected static void clipForOverlappingReads(final SAMRecord read1, final SAMRecord read2, final CigarOperator clippingOperator, final ReadStructure read1Structure, final ReadStructure read2Structure,
+                                                  final List<? extends AdapterPair> adaptersPairs) {
         // If both reads are mapped, see if we need to clip the ends due to small
         // insert size
+        if (!clippingOperator.isClipping()) {
+            throw new PicardException("Cannot use non-clipping operator " + clippingOperator + " to clip overlapping reads");
+        }
         if (!(read1.getReadUnmappedFlag() || read2.getReadUnmappedFlag())) {
             if (read1.getReadNegativeStrandFlag() != read2.getReadNegativeStrandFlag()) {
                 final SAMRecord pos = (read1.getReadNegativeStrandFlag()) ? read2 : read1;
                 final SAMRecord neg = (read1.getReadNegativeStrandFlag()) ? read1 : read2;
 
                 // Innies only -- do we need to do anything else about jumping libraries?
                 if (pos.getAlignmentStart() < neg.getAlignmentEnd()) {
+
+                    if (clippingOperator.equals(CigarOperator.HARD_CLIP)) {
+
+                        //need to consider unclipped positions because often the readthrough bases have already been soft-clipped
+                        final int posClipFrom = getReadPositionToClipFrom(pos, neg.getUnclippedEnd() + 1);
+                        final int negClipFrom = getReadPositionToClipFrom(neg, pos.getUnclippedStart() - 1);
+                        if (posClipFrom == negClipFrom && posClipFrom > 0) {
+                            final int clipFrom = posClipFrom;
+
+                            for (final AdapterPair adapterPair : adaptersPairs) {
+                                if (basesToClipMatchAdapterPair(read1, clipFrom - 1, adapterPair, read2Structure) && basesToClipMatchAdapterPair(read2, clipFrom - 1, adapterPair, read1Structure)) {
+                                    CigarUtil.clip3PrimeEndOfRead(read1, clipFrom, CigarOperator.HARD_CLIP);
+                                    CigarUtil.clip3PrimeEndOfRead(read2, clipFrom, CigarOperator.HARD_CLIP);
+                                    return;
+                                }
+                            }
+                        }
+                    }
+
+
+                    //if we've gotten to here, try standard soft clipping.  Here we consider clipped position because we only are correcting where early sof-clipping isn't quite right
```

Comment:
```suggestion
                    //if we've gotten to here, try standard soft clipping.  Here we consider clipped position because we only are correcting where early soft-clipping isn't quite right
```